# MM&A Supermarket Case

## 1. Exploring the Data (EDA)

In [2]:
# importing necessary libraries
import numpy as np
import pandas as pd

In [3]:
# reading in the csv file into a dataframe
file = 'mma_mart.csv'
df = pd.read_csv(file)

In [4]:
# observing the data structure
df.head()

order_id  product_id                                   product_name  \
0         1       49302                               Bulgarian Yogurt   
1         1       11109  Organic 4% Milk Fat Whole Milk Cottage Cheese   
2         1       10246                          Organic Celery Hearts   
3         1       49683                                 Cucumber Kirby   
4         1       43633           Lightly Smoked Sardines in Olive Oil   

   aisle_id                 aisle  department_id    department  
0       120                yogurt             16    dairy eggs  
1       108  other creams cheeses             16    dairy eggs  
2        83      fresh vegetables              4       produce  
3        83      fresh vegetables              4       produce  
4        95   canned meat seafood             15  canned goods

In [6]:
# finding the shape of the data set (row and column)
df.shape

(987259, 7)

The data shows 987259 rows and 7 columns.

In [7]:
# finding amount of unique entries in each column
df.nunique()

order_id         97833
product_id       35070
product_name     35070
aisle_id           134
aisle              134
department_id       21
department          21
dtype: int64

There are ~98000 orders and ~35000 unique products which is across 134 aisles that belong to 21 departments. This shows that selecting 1000-1200 products for the instabasket aisle is selecting only 1000/35000 = ~3% of the total product selection offered at MM&A supermarket.